var/spack/repos/builtin/packages/libxcb/package.py

```diff
@@ -60,3 +60,8 @@ def configure_args(self):
 
     def patch(self):
         filter_file("typedef struct xcb_auth_info_t {", "typedef struct {", "src/xcb.h")
+
+    def setup_build_environment(self, env):
+        env.set(
+            "LC_ALL", "C.UTF-8"
+        )  # set well defined locale; see: https://www.linuxfromscratch.org/blfs/view/git/x/libxcb.html
```

Comment:
At hopefully the next release this will be fixed upstream and we can accept non-UTF-8 locales again. I added some comments to that effect, including which PRs to track. Are you using a specific locale that you want to include in the comment below and that currently fails?
```suggestion
    # libxcb fails to build with non-UTF-8 locales, see:
    # https://www.linuxfromscratch.org/blfs/view/git/x/libxcb.html
    # https://gitlab.freedesktop.org/xorg/lib/libxcb/-/merge_requests/53 (merged in 1.17.0)
    # https://gitlab.freedesktop.org/xorg/lib/libxcb/-/merge_requests/60
    # If a newer release can be verified to build with LC_ALL=en_US.ISO-8859-1,
    # then we can limit the following function, e.g.
    # when("@:1.17")
    def setup_build_environment(self, env):
        env.set(
            "LC_ALL", "C.UTF-8"
        )
```